mrp_bom_current_stock/reports/report_mrpcurrentstock_xlsx.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+from odoo import models
+from odoo.tools.translate import _
+
+_logger = logging.getLogger(__name__)
+
+
+class ReportMrpBomCurrentStockXlsx(models.AbstractModel):
+    _name = 'report.mrp_bom_current_stock.report_mrpbom_current_stock_xlsx'
+    _inherit = 'report.report_xlsx.abstract'
+
+    @staticmethod
+    def _print_bom_children(ch, sheet, row):
+        i = row
+        sheet.write(i, 0, ch.bom_level or '')
+        sheet.write(i, 1, ch.bom_line.bom_id.code or '')
+        sheet.write(i, 2, ch.product_id.product_tmpl_id.display_name or '')
+        sheet.write(i, 3, ch.product_qty or '')
+        sheet.write(i, 4, ch.product_uom_id.name or '')
+        sheet.write(i, 5, ch.location_id.name or '')
+        sheet.write(i, 6, ch.qty_available_in_source_loc or 0.0)
+        sheet.write(i, 7, ch.bom_id.code or '')
+        sheet.write(i, 8,
+                    ch.bom_id.product_id.product_tmpl_id.display_name or '')
```

Comment:
The template is required, the variant no, this could lead to empty values. Directly product_tmpl_id, remove product_id.